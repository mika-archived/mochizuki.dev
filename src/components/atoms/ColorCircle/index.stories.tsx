import React from "react";
import styled from "styled-components";
import { storiesOf } from "@storybook/react";

import ColorCircle from ".";

const Badge = styled(ColorCircle)`
  width: 24px;
  height: 24px;
`;

storiesOf("atoms/ColorCircle", module).add("default", () => (
  <>
    <Badge color="#178600" />
  </>
));
