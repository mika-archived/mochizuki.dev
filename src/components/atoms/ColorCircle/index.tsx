import React from "react";
import styled from "styled-components";

type Props = {
  className?: string;
  color: string;
};

const Circle = styled.div`
  background-color: ${props => props.color};
  border-radius: 9999px;
`;

const ColorCircle: React.FC<Props> = ({ className, color }) => {
  return <Circle className={className} color={color} />;
};

export default ColorCircle;
