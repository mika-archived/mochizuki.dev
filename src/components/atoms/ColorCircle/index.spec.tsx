import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";

import React from "react";
import { cleanup, render } from "@testing-library/react";

import ColorCircle from ".";

afterEach(cleanup);

test("does not have changes from snapshot", () => {
  const { asFragment } = render(<ColorCircle color="#178600" />);

  expect(asFragment()).toMatchSnapshot();
});
