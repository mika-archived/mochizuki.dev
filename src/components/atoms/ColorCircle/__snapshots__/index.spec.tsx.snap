// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not have changes from snapshot 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #178600;
  border-radius: 9999px;
}

<div
    class="c0"
    color="#178600"
  />
</DocumentFragment>
`;
